log/loggers.go

```diff
@@ -5,122 +5,244 @@ import (
+	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil || !fields.info {
+	if fields == nil {
 		return
 	}
-	displayError(fields.logger.newLogEvent(fmt.Sprintln(v...),
-		fields.logger.InfoHeader,
-		fields.name,
-		fields.output))
+	if fields.info {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintln(v...) },
+			fields.logger.InfoHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Infof takes a pointer subLogger struct, string & interface formats and sends to Info()
+// Infof takes a pointer subLogger struct, string & interface formats
 func Infof(sl *SubLogger, data string, v ...interface{}) {
-	Info(sl, fmt.Sprintf(data, v...))
+	mu.RLock()
+	defer mu.RUnlock()
+	fields := sl.getFields()
+	if fields == nil {
+		return
+	}
+	if fields.info {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintf(data, v...) },
+			fields.logger.InfoHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Debug takes a pointer subLogger struct and string sends to multiwriter
+// Debug takes a pointer subLogger struct and string sends to StageLogEvent
 func Debug(sl *SubLogger, data string) {
+	mu.RLock()
+	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil || !fields.debug {
+	if fields == nil {
 		return
 	}
-	displayError(fields.logger.newLogEvent(data,
-		fields.logger.DebugHeader,
-		fields.name,
-		fields.output))
+	if fields.debug {
+		fields.output.StageLogEvent(func() string { return data },
+			fields.logger.DebugHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Debugln  takes a pointer subLogger struct, string and interface sends to newLogEvent
+// Debugln takes a pointer subLogger struct, string and interface sends to StageLogEvent
 func Debugln(sl *SubLogger, v ...interface{}) {
+	mu.RLock()
+	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil || !fields.debug {
+	if fields == nil {
 		return
 	}
-
-	displayError(fields.logger.newLogEvent(fmt.Sprintln(v...),
-		fields.logger.DebugHeader,
-		fields.name,
-		fields.output))
+	if fields.debug {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintln(v...) },
+			fields.logger.DebugHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Debugf takes a pointer subLogger struct, string & interface formats and sends to Info()
+// Debugf takes a pointer subLogger struct, string & interface formats
```

Comment:
inconsistent comment style
```suggestion
// Debugf takes a pointer subLogger struct, string and interface formats sends to StageLogEvent
```